framework/db/ActiveRelationTrait.php

```diff
@@ -585,22 +585,23 @@ private function getModelKey($model, $attributes)
         if (count($key) > 1) {
             return serialize($key);
         }
-        $key = reset($key);
-        return is_scalar($key) ? $key : serialize($key);
+        return reset($key);
```

Comment:
That was likely something to do with JSON columns that are both a single column in a key and isn't a scalar. I doubt anyone has these but still...

Reply:
So do you think we should keep `normalizeModelKey` to return non-strings as well?

Reply:
No, it won't work. Precisely, this piece will fail immediately:

```php
$key = $this->normalizeModelKey($key);
if (isset($buckets[$key])) {
    if ($this->indexBy !== null) {
        // if indexBy is set, array_merge will cause renumbering of numeric array
        foreach ($buckets[$key] as $bucketKey => $bucketValue) {
            $value[$bucketKey] = $bucketValue;
        }
    } else {
        $value = array_merge($value, $buckets[$key]);
    }
}
```

Reply:
Yes, seems these changes are all correct.